ooi_processing.py

```diff
@@ -1,35 +1,124 @@
+import argparse
 import datetime
 import logging
 import os
 import sys
 
 from ooipy.request import hydrophone_request
 
-from create_spectrogram import save_spectrogram
+from create_spectrogram import create_spec_name, save_spectrogram
 
-logging.basicConfig(
-    format="%(levelname)s:%(message)s", stream=sys.stdout, level=logging.INFO
-)
 
-end_time = datetime.datetime.combine(
-    datetime.datetime.today(), datetime.datetime.min.time()
-)
-start_time = end_time - datetime.timedelta(days=1)
-segment_length = datetime.timedelta(minutes=5)
-node = "PC01A"
+def save_ooi_spectrograms(
+    start_time,
+    end_time,
+    segment_length=datetime.timedelta(minutes=5),
+    node="PC01A",
+    output_dir="spectrograms",
+    NFFT=256,
+):
+    """
+    Creates spectrograms for each time segment in the specified time range.
 
-while start_time < end_time:
-    segment_end = min(start_time + segment_length, end_time)
-    hydrophone_data = hydrophone_request.get_acoustic_data(
-        start_time, segment_end, node, verbose=True
-    )
-    if hydrophone_data is None:
-        logging.info(f"Could not get data from {start_time} to {segment_end}")
+    Args:
+        `start_time`: `datetime.datetime`
+        `end_time`: `datetime.datetime`
+        `segment_length`: `datetime.timedelta`
+        `node`: One of the preset OOI nodes.
+        `output_dir`: Path to the output directory.
+        `NFFT`: The number of data points used in each block for the FFT. A power 2 is most efficient.
+    Returns:
+        None
+    """
+    while start_time < end_time:
```

Comment:
For now this approach is ok, in the future possibly a for loop will help with parallelization (maybe if one gets a runner with more cores)

Reply:
Good idea! I've opened an issue to track it #46